File: src/algorithms/most-frequent.js
/**
 * Given an integer array nums and an integer key, counts the occurrences of
 * numbers that immediately follow key, and returns the number with the most
 * occurrences.
 * 
 * @param {number[]} nums - An array of numbers
 * @param {number} key - A key number in the array
 * @returns {number} - Returns the number with the most occurrences after key
 */
function mostFrequent (nums, key) {
  let count = {};
  for (let i = 0; i < nums.length - 1; i++) {
    if (nums[i] === key) {
      count[nums[i + 1]] = (count[nums[i + 1]] || 0) + 1;
    }
  }
  let max = Object.keys(count).reduce((a, b) => {
    return count[a] > count[b] ? a : b;
  });
  return Number(max);
}

File: src/algorithms/most-frequent.py
def mostFrequent(nums, key):
    """You are given a 0-indexed integer array nums. You are also given an 
    integer key, which is present in nums. For every unique integer target in 
    nums, count the number of times target immediately follows an occurrence of 
    key in nums. In other words, count the number of indices i such that:
    • 0 <= i <= nums.length - 2,
    • nums[i] == key and,
    • nums[i + 1] == target.
    Return the target with the maximum count. The test cases will be generated 
    such that the target with maximum count is unique.

    Args:
        nums (list[int]): A list of integers.
        key (int): An integer.

    Returns:
        int: Returns the maximimum occurring target integer.
    """
    res = [float('-inf'), float('-inf')]
    target = {}
    for i in range(len(nums) - 1):
        if nums[i] == key:
            if nums[i + 1] in target:
                target[nums[i + 1]] += 1
            else:
                target[nums[i + 1]] = 1
    for key, val in target.items():
        if val > res[1]:
            res = [key, val]
    return res[0]
